### PR TITLE
docs(status-list): add module readme

### PR DESCRIPTION
**Pull request by jrhender**
Adding documentation on `status-list` design
https://github.com/energywebfoundation/ssi-hub/blob/status-list-documentation/docs/status-list.md

### CODE REVIEW

**Review by Harasz — APPROVED** [review]

File: docs/status-list.md [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+### Revocation Authorization is based on Role Definition
+It only makes sense for a revoker to sign a list that they are authorized to sign.
+This authorization is determined by the role definition of the role credential that a revoker is attempting to revoke.
+
+This means that it is not feasible have just one list in SSI Hub, for example.
+This is because it is not feasible that a given revoker is authorized to revoke all roles.
+
+**Chosen solution**:
+
+This authorization challenge is eliminated by having one list per role definition/namespace.
```
Comment (JGiter) [inline docs/status-list.md:65]: Isn't status list created for each credential? https://github.com/energywebfoundation/ssi-hub/blob/d1b42b2f434e8e2251c3d1a2b875bde4f317c9af/src/modules/status-list/entities/namespace-status-lists.entity.ts#L49

File: docs/status-list.md [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+
+This approach allows bulk revocation of credential of the same role
+
+### Client-side proving/signing of CredentialStatus VC
+[StatusList2021Credentials](https://w3c-ccg.github.io/vc-status-list-2021/#example-example-statuslist2021credential-0) should have a proof so that verifiers can con
+
+Due to the IAM stack's signature being client-side (performed, for example by Metamask) means that the server cannot forge status list credentials.
+However this introduces challenges because the revocations cannot be performed synchronously by the server.
+
+The challenge is as follows:
```
Comment (JGiter) [inline docs/status-list.md:80]: There is something missing here

File: docs/status-list.md [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+
+
+Therefore, only one revoker may update a given StatusList2021Credential at a given time.
+
+**Chosen solution:**
+
+Have a 1to1 mapping of StatusListCredential to VC.
```
Comment (JGiter) [inline docs/status-list.md:87]:
```suggestion
Have a 1 to1 mapping of StatusListCredential to VC.
```

File: docs/status-list.md (hunk 1 above)
Comment (jrhender) [reply docs/status-list.md:65]: Good point, I changed the wording to be "having the credentials on each given list be only for one role."

File: docs/status-list.md (hunk 2 above)
Comment (jrhender) [reply docs/status-list.md]: Thanks, I added a mermaid diagram

**Review by JGiter — APPROVED** [review]

**Pull request merged**